miniapp/miniapp_cholesky.cpp

```diff
@@ -237,13 +237,14 @@ void setUpperToZeroForDiagonalTiles(Matrix<T, Device::CPU>& matrix) {
     if (distribution.rankIndex() != distribution.rankGlobalTile(diag_tile))
       continue;
 
-    auto tile_set = unwrapping([](auto&& tile) {
+    auto tile_set = [](typename Matrix<T, Device::CPU>::TileType&& tile) {
```

Comment:
In some cases it's required. One-argument callables get instantiated internally in pika in an unrelated path with an unrelated type and fails to compile with `auto`. It's on my todo list to see if we can avoid that instantiation (though now I'm reminded to open an issue). In other cases it's just because I find it nice to be explicit but it's not strictly required.

Reply:
For reference, this is the corresponding issue: https://github.com/pika-org/pika/issues/181.